### DISCUSSION TITLE
Eco-friendly "offline" integrations support

### DISCUSSION THREAD

**Discussion by andriej** [home-assistant/architecture#629] [General]
Lately there has been Energy added to HA which indicated some energy-saving movement throughout HA users (inlucing myself).
I've noticed that I don't use NAS that much and it runs 24/7, so I added automation to let it suspend/go to sleep when not needed and wake-on-lan power up when needed.

That showed one issue - Synology NAS integration (it's just example, not issue with it) is meant to be 100% time online. 
In my opinion, just as `media_player` integrations including some TV - there should be possibility for integration to be "actively managed offline", so if HA can't connect to it - it should be shout out as error, but as a well-known value because user meant to make it offline on purpose.

As UI and logs matter quite a lot for me, seeing this:
![image](https://user-images.githubusercontent.com/6303489/131731562-d156f2af-d8da-4ce6-8aa6-429e3051c1c0.png)

means there's something wrong with integration, while everything is fine - it's just offlline.

It's more UI-issue I guess, but need some architectural changes too to let users have different integrations with their devices (printers, NAS, climate [hvac on winter]) powered off without error thrown everywhere.

I hope I've stated it clear, that it's not clearly integration-issue and more like "eco friendly, offline-capable integration". :-)

**Comment by MartinHjelmare** [comment]
This sounds like a feature request and not an architecture change suggestion. Please only post here if you have a concrete suggestion for a change and you're willing to implement that change. Answer what, why and how.

**Reply by andriej** [reply]
Is maintainer of this example integration able to declare it as 'partially offline' so it doesn't throw errors like that when it's end device is literally offline on purpose?
I thought it's architecturally capable only for media_players.

If it is, then it is more feature request, right. But I'm not sure backend/architecture is ready for that.

**Reply by MartinHjelmare** [reply]
Any integration can be set up while ignoring connection errors. If that is a good idea, is another question. Each integration will need to adjust this depending on what kind of device or service is integrated. Entities use the `available` property to mark if its corresponding device or service is online/offline.

**Comment by frenck** [comment]
Closing this architectural discussion, as it seems to be an FR, not an architectural proposal.

../Frenck
